scripts/eval-pr

```diff
@@ -0,0 +1,193 @@
+#!/bin/bash
+#---
+# eval-pr
+#
+# Copyright (c) 2022 Freeciv21. This file is part of Freeciv21. Freeciv21 is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the  License, or (at your option) any later version. You
+# should have received  a copy of the GNU General Public License along with Freeciv21. If not, see
+# https://www.gnu.org/licenses/.
+#
+# Bash shell script to aid the process of evaluating a pull request
+#--
+
+version="1.0"
+re='^[0-9]+$'
+
+if [[ $2 ]]; then
+  pull_request=$2
+fi
+
+help ()
+{
+  echo "Freeciv21: Evaluate a Pull Request as documented at"
+  echo "  https://longturn.readthedocs.io/en/latest/Contributing/eval-pull-request.html"
+  echo "Version: $version"
+  echo ""
+  echo "Should be executed from the root of the Freeciv21 source directory."
+  echo ""
+  echo "Syntax: ./scripts/eval-pr -h|s|a|c"
+  echo ""
+  echo "-h                Opens this help"
+  echo "-s [PR Number]    Starts a new full evaluation process"
+  echo "-a [PR Number]    Starts a new evaluation process and also run an autogame through ASAN"
+  echo "-c [PR Number]    Cleanup a previously run evaluation process"
+  echo ""
+}
+
+base_setup ()
+{
+  local branch=$(git rev-parse --abbrev-ref HEAD)
+  if ! [[ "$branch" == "master" ]]; then
+    echo "-- Stashing changes and checking out master branch."
+    echo ""
+    git stash
+    git checkout master
+    echo ""
+  fi
+
+  echo "-- Fetching upstream master"
+  echo ""
+  git fetch upstream master
+
+  echo ""
+  echo "-- Creating a testing branch for PR $pull_request."
+  echo ""
+  git checkout -b testing/pr_$pull_request upstream/master
+
+  echo ""
+  echo "-- Getting a .diff of PR $pull_request."
+  echo ""
+  wget https://github.com/longturn/freeciv21/pull/$pull_request.diff -O pr$pull_request.diff
+
+  echo ""
+  echo "-- Applying the .diff via patch."
+  echo ""
+  patch -p1 < pr$pull_request.diff
+  rm pr$pull_request.diff
+  rm -Rf build
```

Comment:
Noo you nuked my build directory!

Reply:
Yea I like @pranavtifr idea to do a special purpose build directory to leave the standard one alone.